ui/de-lib/src/main/java/org/iplantc/de/admin/desktop/client/toolAdmin/presenter/ToolAdminPresenterImpl.java

```diff
@@ -0,0 +1,141 @@
+package org.iplantc.de.admin.desktop.client.toolAdmin.presenter;
+
+import org.iplantc.de.admin.desktop.client.toolAdmin.ToolAdminView;
+import org.iplantc.de.admin.desktop.client.toolAdmin.service.ToolAdminServiceFacade;
+import org.iplantc.de.client.models.tool.ToolAutoBeanFactory;
+import org.iplantc.de.client.models.tool.Tool;
+import org.iplantc.de.client.models.tool.ToolList;
+import org.iplantc.de.commons.client.ErrorHandler;
+import org.iplantc.de.commons.client.info.IplantAnnouncer;
+import org.iplantc.de.commons.client.info.SuccessAnnouncementConfig;
+
+import com.google.gwt.user.client.rpc.AsyncCallback;
+import com.google.gwt.user.client.ui.HasOneWidget;
+import com.google.inject.Inject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by aramsey on 10/27/15.
+ */
+
+
+public class ToolAdminPresenterImpl implements ToolAdminView.Presenter {
+
+    private final ToolAdminView view;
+    private final ToolAdminServiceFacade toolAdminServiceFacade;
+    private final ToolAutoBeanFactory factory;
+    private final ToolAdminView.ToolAdminViewAppearance appearance;
+
+    @Inject
+    public ToolAdminPresenterImpl(final ToolAdminView view,
+                                  ToolAdminServiceFacade toolAdminServiceFacade,
+                                  ToolAutoBeanFactory factory,
+                                  ToolAdminView.ToolAdminViewAppearance appearance) {
+        this.view = view;
+        this.factory = factory;
+        this.appearance = appearance;
+        view.setPresenter(this);
+        this.toolAdminServiceFacade = toolAdminServiceFacade;
+    }
+
+    @Override
+    public void go(HasOneWidget container) {
+        container.setWidget(view);
+        updateView();
+
+    }
+
+    @Override
+    public void addTool(Tool tool) {
+
+        //The UI handles creating a single tool request, but the admin/tools POST endpoint requires
+        // an array of requests.  Wrapping the request inside an array.
+        ToolList toolListAutoBean = factory.getToolList().as();
+        List<Tool> toolList = new ArrayList<Tool>();
+        toolList.add(tool);
+        toolListAutoBean.setToolList(toolList);
+
+        toolAdminServiceFacade.addTool(toolListAutoBean, new AsyncCallback<Void>() {
+
+            @Override
+            public void onFailure(Throwable caught) {
+                ErrorHandler.post(caught);
+            }
+
+            @Override
+            public void onSuccess(Void result) {
+                IplantAnnouncer.getInstance().schedule(new SuccessAnnouncementConfig(appearance.addToolSuccessText()));
+                updateView();
+            }
+
+        });
+    }
+
+    @Override
+    public void updateTool(Tool tool) {
+        toolAdminServiceFacade.updateTool(tool, new AsyncCallback<Void>() {
+            @Override
+            public void onFailure(Throwable caught) {
+                ErrorHandler.post(caught);
```

Comment:
The `POST /admin/tools` endpoint will return an `ERR_NOT_WRITEABLE` if the admin attempts to update container settings for a "public" tool (tools used in public apps), but we don't want to completely prevent admins from updating a public tool's container settings, so the endpoint provides an `overwrite-public` flag that may be included in the request to update these settings for public tools.

The UI is intended to use this flag in a similar way it's using the `force` flag with the copy metadata endpoint: see the [CopyMetadataCallback](https://github.com/iPlantCollaborativeOpenSource/DE/blob/dev/ui/de-lib/src/main/java/org/iplantc/de/diskResource/client/presenters/grid/GridViewPresenterImpl.java#L125-L171) for an example.

I would also encourage you to use a message similar to the `Danger Zone` message of the `POST /admin/tools` endpoint when presenting the admin an `overwrite-public` confirmation dialog.